File: hashmap/twoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    var map = {};
    for (var i=0;i<nums.length;i++){
        if(map[nums[i]]!==undefined){
            return [map[nums[i]],i];//return i and stored index
        }else{
            //store index
            map[target-nums[i]] =i;//if another half which add to target found return
        }
    }
};

File: hashmap/twoSum/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        record = {}
        for i,num in enumerate(nums):
            if target - num in record:
                return [record[target-num],i] # if target - num exist return i and previous another part add to target
            else:
                record[num] = i #index for itself
'''
enumerate will do
for i,num in enumerate(project):
    print(i,num)
    #######
    1 apple
    2 PC
    3 Mobile
    ...
'''
